src/app/shared/components/template/components/map/map.component.ts

```diff
@@ -0,0 +1,203 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import { TemplateAssetService } from "../../services/template-asset.service";
+import { getParamFromTemplateRow, getStringParamFromTemplateRow } from "src/app/shared/utils";
+import { AppDataService } from "src/app/shared/services/data/app-data.service";
+import Map from "ol/Map";
+import View from "ol/View";
+import { OSM } from "ol/source";
+import TileLayer from "ol/layer/Tile";
+import VectorSource from "ol/source/Vector";
+import VectorLayer from "ol/layer/Vector";
+import GeoJSON from "ol/format/GeoJSON";
+import Style from "ol/style/Style";
+import Fill from "ol/style/Fill";
+import Stroke from "ol/style/Stroke";
+import chroma from "chroma-js";
+import BaseLayer from "ol/layer/Base";
+
+interface IMapLayer {
+  id: string;
+  description: string;
+  fill: string;
+  name: string;
+  opacity: number;
+  // a property of the dataset to be plotted
+  property: string;
+  scale_fill: string;
+  scale_max: number;
+  scale_min: number;
+  // the path to the GeoJSON file containing the data to be plotted
+  source_asset: string | any;
+  stroke: string;
+  type: "vector" | "heatmap";
+  visible_default: boolean;
+}
+
+interface IMapParams {
+  /**
+   * TEMPLATE PARAMETER: extent.
+   * Optional. The extent of the map to be displayed, i.e. the boundaries of the visible map, in EPSG:3857 co-ordinates
+   */
+  extent: number[];
```

Comment:
nit(non-blocking): I'm assuming this terminology is to maintain consistency with OL, but might be worth making more human readable and including an example. E.g. `boundToCoordinates` or similar. I'd also recommend an option to automatically generate bounds based on data in the future (e.g. `boundToData: true`), or possibly the extent of any geojson layers provided